receiver/pluginreceiver/plugin.go

```diff
@@ -91,6 +100,32 @@ func (p *Plugin) ApplyDefaults(values map[string]any) map[string]any {
 	return result
 }
 
+// CheckExtensions checks to see if there exists any filestorage extensions
+// within the plugin config. If one exists, make a unique storage extension
+// directory based on the receiver component ID.
+func CheckExtensions(values map[string]any, id string) error {
+	for i, ext := range values {
+		if i == "file_storage" {
+			var cfg filestorage.Config
+			err := mapstructure.Decode(ext, &cfg)
+			if err != nil {
+				return err
+			}
+
+			name := strings.Replace(id, "/", "_", -1)
```

Comment:
Should use [strings.ReplaceAll](https://pkg.go.dev/strings#ReplaceAll) as it's the same as `-1`.